### ISSUE TITLE
Regression in Proton 8 causing slow launch under certain conditions

### ISSUE THREAD

**Issue by simifor** [ValveSoftware/Proton#7332]
I first noticed this issue last month, but I was unable to find what triggered the issue until today. I tested 8.0.4, 8.0.5-RC and experimental, all of which were affected.

# Effect

My baseline is Proton 7, and the tested games are the following:
- Victory Heat Rally (Demo): Launch time goes from ~5 seconds to ~125.
- Darksiders (not Warmastered): Launch times goes from ~10 seconds to ~131.
- Snake Pass*: Launch time goes from ~26 seconds to ~147.
- Mortal Shell: Launch time goes from ~12 seconds to ~133.

The times were very consistent, with a 1-second delta after several tests, and the difference seems to always be around 2 minutes, so I think it might be some sort of timeout.

*Snake Pass behaves a bit differently from the rest, as it will always launch a window around the ~25 second mark with and without the bug. But after the bug has triggered that screen will remain black and not progress until roughly 2 minutes have passed vs instantly starting playing the startup videos when the bug hasn't been triggered.

Darksiders is installed in an NTFS partition on an nvme SSD, the rest are installed on an EXT4 on an HDD, so filesystem and installation medium don't seem to be a significant factor.

# Trigger

I used Victory Heat Rally as the coal canary as it has the shortest launch time. The only way I found to trigger the issue has been to **play Dirt Showdown**, closing it immediately after the game launches seems enough to activate the problem. There might be other ways to cause this issue, but this is the one that has "worked" for me thus far.

I think that Dirt Showdown uses CEG, and think it might be related, but it could just be a coincidence. After the issue becomes visible on Victory Heat Rally, it also becomes apparent in the other games mentioned.

After the effect is in place, the only way I've found to get rid of it is to reboot the computer., or switching to Proton 7 which is unaffected

# Things that didn't trigger it (and don't make the effect go away)

- Launching Victory Heat Rally several times.
- Closing/reopening steam.
- Changing Proton versions.
- Launching native games.
- Launching affected games again.
- Verifying game files.
- Deleting the game's prefix.
- Disabling ESYNC and FSYNC.
- logging out of my session.

# Logs

[steam-1739050-RCfast.log](https://github.com/ValveSoftware/Proton/files/13629481/steam-1739050-RCfast.log)
[steam-1739050-RCslow.log](https://github.com/ValveSoftware/Proton/files/13629482/steam-1739050-RCslow.log)
RCfast is Victory Heat Rally before the bug gets triggered. Unfortunately, I don't think they'll be useful, as I didn't find any meaningful difference between them.

System info: https://gist.github.com/simifor/4cb5b3852884dec2519323cfb55eaeb8

**Comment by simifor** [comment]
I'm closing this issue as it doesn't seem to be a problem with proton itself. Upon further testing, Dirt Showdown only causes issues for other games when it is installed to my bcachefs partition. This doesn't explain the slowdowns I noticed in November as the game was in my EXT4 partition at the time, but at this point I haven't found a way to reproduce this issue on demand without involving bcachefs.

**Comment by simifor** [comment]
I can still reproduce the issue by launching dirt showdown, additionally I found some more games that have their launch time negatively impacted:
- Void Stranger
- Anger foot (demo)
- Exanima
- The murder of sonic

Edit: Mortal shell is affected, my testing methodology made me think otherwise. Every other game would either take a long time to launch a window or launch a window early and then take a lot of time to render anything, so when mortal shell launched rendering a spinner I thought it was working fine, but if you actually wait for the game to load you can see that it's taking much longer.

So at this point, every game I have tried at this point is negatively affected by this bug on proton 8.

**Comment by alasky17** [comment]
@simifor I'm not seeing this locally ... I have a few questions - 
- Dirt Showdown looks like it has a Linux version - do you have it manually forced to use Proton? 
- How are you closing Dirt Showdown - are you waiting for the window to show up, the main menu, etc?  Are you using alt+f4?  If so, could you try closing the game using the in-game menu and see if you still see this issue? 
- "I first noticed this issue last month" - does that mean that it was not happening on earlier versions of Proton 8.0-x?  I'm curious because Proton 8.0-x has been out for much longer than that.  Was there anything else in your setup that changed? 
- Do you have any extra stuff impacting the way Proton games run - mangohud, gamemode, etc?  If so, could you disable these and make sure the problem still happens?

**Comment by simifor** [comment]
> @simifor I'm not seeing this locally ... I have a few questions

Yeah, unfortunately Gofman couldn't reproduce it himself either, so it must be configuration dependant.

>     * Dirt Showdown looks like it has a Linux version - do you have it manually forced to use Proton?
yes

>     * How are you closing Dirt Showdown - are you waiting for the window to show up, the main menu, etc?  Are you using alt+f4?  If so, could you try closing the game using the in-game menu and see if you still see this issue?
I usually use alt+f4, but exiting through "exit showdown" causes the same effect.

>     * "I first noticed this issue last month" - does that mean that it was not happening on earlier versions of Proton 8.0-x?  I'm curious because Proton 8.0-x has been out for much longer than that.  Was there anything else in your setup that changed?
I don't boot dirt showdown often, so it's more likely that I didn't run it with earlier versions, though proton GE 8-1 shows the same issue, so I think it's likely it would trigger with older proton 8 versions. The only things that have changed in my setup is changing my 380x card with a 6600 card and adding a NVME SSD, but I didn't test the game close to these changes so I couldn't say if it's related

>     * Do you have any extra stuff impacting the way Proton games run - mangohud, gamemode, etc?  If so, could you disable these and make sure the problem still happens?
I don't use gamemode and mangohud seems to have no effect. I've usually ran the games with just "%command%" in launch parameters and noticed no difference in launch times.

**Comment by alasky17** [comment]
@simifor Thank you for the information :) Unfortunately I don't think I'll be of much help since I could not repro and gofman already took a look and didn't see anything :(  It sounds like something on your system that would need to be narrowed down locally, which is rather hard to do without another system to compare similarities/differences.

**Comment by simifor** [comment]
Yeah, I don't think it's worth looking into this with the available information. If I manage to find something else I'll post it here, and if someone else suffers from it, hopefully they'll find this issue.

**Comment by simifor** [comment]
It seems that this issue is related to audio, changing from pulse to alsa, or disabling audio outright, make games launch fast again as expected. Setting it back to pulse will slow down game launch again.

Edit:

I have found another way to trigger the issue, some conditions needed for it to happen, and a workaround once it happens. Besides dirt showdown, I can trigger the slowdown by launching the native version of the capes demo, you don't need to play, you can just quit when the menu loads.

Other conditions:
- Having wireplumber installed, I couldn't get the issue to appear when using pipewire-media-session.
- Having my IFPU turned on as an audio device. My computer has 3 audio devices, the motherboard itself, the IGPU, and the dedicated GPU. IGPU needs to be turned on as an audio device (not necessarily in use) while playing one of the problematic games for the slowdown to trigger. 

Workaround:
After the slowdown is active, turning off the IGPU as an audio device will "untrigger" it, after this you can turn on the IGPU again without the issue returning unless you play one of the problematic games again.

**Comment by ninetailedtori** [comment]
I have this issue in new Protons as well, actually. Wireplumber repro here.
